pyqir/tests/test_value_operators.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from typing import List, Optional, Tuple
+
+from pyqir import (
+    Context,
+    Instruction,
+    Module,
+    Value,
+)
+import pytest
+
+
+def get_module() -> Module:
+    llvm_ir = """
+    define void @program_main() {
+    entry:
+    %0 = add i64 1, 2
+    %1 = mul i64 %0, %0
+    ret void
+    }
+    """
+
+    mod = Module.from_ir(Context(), llvm_ir, "module")
+    return mod
+
+
+def get_instructions(module: Optional[Module] = None) -> List[Instruction]:
+    mod: Module = module if module else get_module()
+    return mod.functions[0].basic_blocks[0].instructions
+
+
+def get_first_instruction_operands() -> Tuple[Value, Value]:
+    insts = get_instructions()
+    operand10: Value = insts[1].operands[0]
+    operand11: Value = insts[1].operands[1]
+    return (operand10, operand11)
+
+
+def test_instruction_equals_instruction() -> None:
+    inst = get_instructions()[0]
+    assert inst == inst
+
+
+def test_instruction__eq__instruction() -> None:
+    inst = get_instructions()[0]
+    assert inst.__eq__(inst)
+
+
+def test_instruction_is_self_instruction() -> None:
+    inst = get_instructions()[0]
+    assert inst is inst
+
+
+def test_instruction_not_equals_instruction() -> None:
+    insts = get_instructions()
+    assert insts[0] != insts[1]
+
+
+def test_instruction__ne__instruction() -> None:
+    insts = get_instructions()
+    assert insts[0].__ne__(insts[1])
+
+
+def test_instruction_is_not_other_instruction() -> None:
+    insts = get_instructions()
+    assert insts[0] is not insts[1]
+
+
+def test_instruction_hash_equals_self_hash() -> None:
+    inst = get_instructions()[0]
+    assert hash(inst) == hash(inst)
+
+
+def test_instruction_hash_not_equals_other_hash() -> None:
+    insts = get_instructions()
+    assert hash(insts[0]) != hash(insts[1])
```

Comment:
I'm not sure I would test this. It's not a guaranteed property based on the interface, even though it's probably the case that the hasher just copies the pointer into the hash on 64-bit systems.

Reply:
You're right.